### ISSUE TITLE
Power Down Device

### ISSUE THREAD

**Issue by merkste** [Zyg0te/Locker#12]
Instead of wiping all data I'd like to see an option to just power down the device. This should be safe given encryption is enabled and set-up with some non-trivial password.

**Comment by Zyg0te** [comment]
I agree. This would be useful.

**Comment by Zyg0te** [comment]
It turns out this is a bit more complicated than I first anticipated. Powering off the device doesnt seem to be possible at all for third party apps. 

The alternative is to reboot the device. This can be accomplished with PowerManager.
```

PowerManager pm = (PowerManager) context.getSystemService(Context.POWER_SERVICE);
                    pm.reboot(null);
```

However, this requires the REBOOT permission which is only granted to system apps. 
The DevicePolicyManager does not include the ability to reboot the device, though Android N [seemingly ads that ability](http://developer.android.com/preview/features/afw.html).

**Comment by Micha-Btz** [comment]
i don't have tested it, but maybe this works.
https://github.com/mkodekar/WrongPinShutdown/blob/master/src/com/pikselbit/wrongpinshutdown/LogInReceiver.java#L37

**Comment by Zyg0te** [comment]
If you look at that code:

`new String[] { "su", "-c", "reboot", "-p" });`

It's using a shell command that requires root (su). That works fine on rooted devices but that's not what we are looking for here unfortunately.

Thanks.

**Comment by Micha-Btz** [comment]
thats true but for people with rooted devices is it a working solution.
if someone try to unlock my device they will have another chance it the device just rebooted.
so in my opinion reboot is not very usefull.

**Comment by Zyg0te** [comment]
Indeed. The usecase for a reboot feature is simply that if you have device encryption enabled and the device reboots, user data will be encrypted and not easily recoverable by an attacker unless he's able to figure out the unlock code.

**Comment by fowlslegs** [comment]
I'm too in favor of giving this option to rooted users. The app would not need to prompt for root unless the user wanted to use this option and you could have the menu entry (and docs) say `shutdown (requires root)` to make that requirement clear. When Android N roles around, unrooted users who are running it would be able to have this option as well, while still allowing for rooted users of older Android versions to use it.

**Comment by Zyg0te** [comment]
Thanks for the input. I'll contemplate it.
